src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/NodeLabelMappingToLiteralsImpl.java

```diff
@@ -0,0 +1,24 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+
+public class NodeLabelMappingToLiteralsImpl implements NodeLabelMapping{
+
+    @Override
+    public Node map(final String label) {
+        return NodeFactory.createLiteral(label);
+    }
+
+    @Override
+    public String unmap(Node node) {
+        if (!node.isLiteral())
+            throw new IllegalArgumentException("The given RDF term (" + node.toString() + ") is not a literal.");
+        return node.getLiteral().toString();
```

Comment:
Instead of using the toString() function, return the lexical form of the literal. There should be a function for that in the `Node` interface of Jena.